R/diffResult.R

```diff
@@ -0,0 +1,194 @@
+            )
+        },
+        
+        add_checkResult = function(CR){
+            self$checks[[CR$name]] <- CR
+        },
+        
+        generate_summary = function(){
+            
+            if(is.null(self$keys)){
+                keys <- "  None Provided  "
+            } else {
+                keys <- self$keys
+            }
+            
+            keysum <- data.frame(
+                "Variable" = keys,
+                stringsAsFactors = FALSE
+            )
+            
+            class_base <- as_cropped_char( paste0(self$meta$class_base, collapse = ", "))
+            class_comp <- as_cropped_char( paste0(self$meta$class_comp, collapse = ", "))
+            
+            
+            dfsum <-  data.frame(
+                "Summary" = c("Number of rows", "Number of columns", "Class"),
+                "Base" = c( self$meta$nrow_base, self$meta$ncol_base, class_base),
+                "Compare" = c( self$meta$nrow_comp, self$meta$ncol_comp, class_comp),
+                stringsAsFactors = FALSE
+            )
+            
+            results <- data.table(
+                Name = character(0),
+                Result = character(0)
+            )
+            for( i in self$checks){
+                results <- rbind(results, data.table(
+                    Name = i$name,
+                    Result = i$result
+                ))
+            }
+            
+            self$check_summary = results
+            
+            if(self$result == "Passed"){
+                line <- "All checks have passed"
+            } else {
+                line <- "Not all checks have passed, details are provided below"
+            }
+            
+            self$summary <- display$new(
+                title = sprintf("Comparison of Base vs Compare"),
+                body = list(
+                    "Dataset Summary",
+                    dfsum,
+                    "",
+                    "Listing of Keys",
+                    keysum,
+                    "",
+                    "Check Summary",
+                    results,
+                    "",
+                    line,
+                    ""
+                )
+            )
+        }
+    )
+)
+
+
+
+
+#' print.diffResult
+#' 
+#' Print method for a diffdf result object (diffResult). Can be rendered
+#' into different output formats and saved into a report file.
+#' 
+#' @param x diffResult Object
+#' @param type Output format. Options = "ascii" (default) or "html"
+#' @param rowlimit How many rows of a display dataset should be shown (default = 10)
+#' @param file Location and name of a text file to output the results to. Setting to NULL will cause no file to be produced.
+#' @param display Whether to print the rendered output in the R console
+#' @param ... Additional arguments (not used)
+#' 
+#' @export
+print.diffResult <- function(
+    x, 
+    type = "ascii", 
+    rowlimit = 10, 
+    file = NULL, 
+    display = is.null(file), 
+    ...
```

Comment:
option to suppress summary would be good.

Reply:
Addressed with c092cd0

Also added in `match.call()` to display the dataset names in the header which should hopefully be more meaningful i.e.

![image](https://user-images.githubusercontent.com/8447209/104122369-8c2f8600-533c-11eb-999f-54a7e52a1d4e.png)